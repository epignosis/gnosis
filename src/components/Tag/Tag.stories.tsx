import React from "react";
import { Story } from "@storybook/react";
import Tag, { TagProps } from "./Tag";

export default {
  component: Tag,
  title: "Components/Tag",
  argTypes: {
    bgColor: {
      control: {
        type: "select",
        options: ["course", "card", "new"],
      },
    },
  },
};

export const Default: Story<TagProps> = (args) => <Tag {...args} />;

Default.args = {
  bgColor: "course",
  children: "This is a tag",
};
