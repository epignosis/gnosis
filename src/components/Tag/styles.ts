import { css, Theme, SerializedStyles } from "@emotion/react";

export const tag = ({ typeScaleSizes }: Theme): SerializedStyles => css`
  display: inline-flex;
  align-items: center;
  font-size: ${typeScaleSizes.xs};
  padding: 0.5rem 0.75rem 0.4rem;
`;
