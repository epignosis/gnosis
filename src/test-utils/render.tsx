import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
