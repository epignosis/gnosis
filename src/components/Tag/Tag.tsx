import React, { FC } from "react";
import { tag } from "./styles";

export type TagProps = React.HTMLAttributes<HTMLSpanElement>;

const Tag: FC<TagProps> = ({
  children,
  style = { backgroundColor: "#a3dfe3", color: "#5c5c5c" },
  ...rest
}) => (
  <span css={tag} style={style} {...rest}>
    {children}
  </span>
);

export default Tag;
