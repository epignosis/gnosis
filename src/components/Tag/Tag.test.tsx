import React from "react";
import faker from "faker";
import Tag from "./Tag";
import { render, screen } from "@test-utils/render";

describe("<Tag />", () => {
  it("renders", () => {
    const tagTxt = faker.lorem.word();
    render(<Tag>{tagTxt}</Tag>);

    const tag = screen.getByText(tagTxt);

    expect(tag).toHaveTextContent(tagTxt);
  });

  it("matches snapshot", () => {
    const { container } = render(
      <Tag style={{ backgroundColor: "purple", color: "white" }}>This is a tag</Tag>,
    );

    expect(container).toMatchSnapshot();
  });
});
