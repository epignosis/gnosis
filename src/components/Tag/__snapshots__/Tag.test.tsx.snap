// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> matches snapshot 1`] = `
<div>
  <span
    class="tag css-3yrqyb-tag-tag"
    id="tag-1"
    style="background-color: purple; color: white;"
  >
    This is a tag
  </span>
</div>
`;
